docs/src/visualization.md

```diff
@@ -0,0 +1,185 @@
+plotted by the `error_metric="metric-name"` keyword argument, . See
+[`compute_error!`](@ref) for a list of the computed error metrics and their
+names.
+
+```@example makie_plots
+# Assume that ekp is the EnsembleKalmanProcess object
+using CairoMakie # load a Makie backend
+import EnsembleKalmanProcesses.Visualize as viz
+
+fig_errors = CairoMakie.Figure(size = (300 * 2, 300 * 1))
+viz.plot_error_over_iters(fig_errors[1, 1], ekp, color = :tomato)
+# Error plotting functions support plotting different errors through the
+# error_metric keyword argument
+ax1 = CairoMakie.Axis(fig_errors[1, 2], title = "Average RMSE over iterations", yscale = Makie.pseudolog10)
+viz.plot_error_over_time!(
+    ax1,
+    ekp,
+    linestyle = :dashdotdot,
+    error_metric = "avg_rmse",
+)
+fig_errors
+```
+
+### Plot constrained parameters
+
+The functions `plot_ϕ_over_iters` and `plot_ϕ_over_time` and the mutating
+versions can be used to plot a scatter plot of an individual parameter (dimension) over
+the number of iterations or time respectively. All keyword arguments supported
+by [`Makie.Scatter`](https://docs.makie.org/dev/reference/plots/scatter) are
+supported by these functions too.
+
+Furthermore, there are also `plot_ϕ_mean_over_iters` and `plot_ϕ_mean_over_time`
+and the mutating versions can be used to plot the mean parameter (dimension)
+over the number of iterations or time respectively. All keyword arguments
+supported by [`Makie.Scatter`](https://docs.makie.org/dev/reference/plots/lines)
+are supported by these functions too. Furthermore, there is the keyword argument
+`plot_std = true` which can be used to also plot the standard deviation as well.
+To support plotting both mean and standard deviation, there are also the keyword
+arguments `line_kwargs` and `band_kwargs` for adjusting how the mean and
+standard deviation are plotted. Any keyword arguments accepted by
+(`Makie.Lines`)[[https://docs.makie.org/dev/reference/plots/lines]] can be
+passed to `line_kwargs` and any keyword arguments accepted by
+(`Makie.Band`)[https://docs.makie.org/dev/reference/plots/band] can be passed to
+`band_kwargs`.
+
+```@example makie_plots
+using CairoMakie # load a Makie backend
+import EnsembleKalmanProcesses.Visualize as viz
+
+fig_ϕ = CairoMakie.Figure(size = (300 * 2, 300 * 2))
+EnsembleKalmanProcesses.Visualize.plot_ϕ_over_time(fig_ϕ[1, 1], ekp, prior, 1, axis = (xscale = Makie.pseudolog10,))
+ax1 = CairoMakie.Axis(fig_ϕ[1, 2])
+viz.plot_ϕ_mean_over_iters!(ax1, ekp, prior, 2)
+viz.plot_ϕ_mean_over_iters(
+            fig_ϕ[2, 1],
+            ekp,
+            prior,
+            3,
+            plot_std = true,
+            line_kwargs = (linestyle = :dash,),
+            band_kwargs = (alpha = 0.2,),
+            color = :purple,
+        )
+fig_ϕ
+```
+
+#### Plot by distribution name
+
+There are also convenience functions for plotting the constrained parameters
+belonging to a distribution. These are `plot_ϕ_over_iters_by_dist`,
```

Comment:
The phrase "constrained parameters belonging to a distribution" does not really make sense. Perhaps we can rephrase to 
 "for plotting the constrained (possibly-multidimensional) parameters by name"